cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/vertx/EndPoints.java

```diff
@@ -0,0 +1,366 @@
+                            Map<String, Object> parameterConfigOverrides)
+            throws Exception {
+        VertxHttpFrameworkHandler handler = new VertxHttpFrameworkHandler(context);
+        CruiseControlParameters parameters;
+        RequestParameterWrapper requestParameter = requestParameterFor(endPoint);
+        if (endPoint == REVIEW) {
+            // Sanity check: if the request is for REVIEW, two step verification must be enabled.
+            if (!_twoStepVerification) {
+                throw new ConfigException(String.format("Attempt to access %s endpoint without enabling '%s' config.",
+                        endPoint, WebServerConfig.TWO_STEP_VERIFICATION_ENABLED_CONFIG));
+            }
+
+            parameters = _config.getConfiguredInstance(requestParameter.parametersClass(), CruiseControlParameters.class, parameterConfigOverrides);
+            if (!hasValidParameterNames(handler, _config, parameters)) {
+                return;
+            }
+        } else if (!_twoStepVerification) {
+            // Do not add to the purgatory if the two-step verification is disabled.
+            parameters = _config.getConfiguredInstance(requestParameter.parametersClass(), CruiseControlParameters.class, parameterConfigOverrides);
+            if (!hasValidParameterNames(handler, _config, parameters)) {
+                return;
+            }
+        } else {
+            // Add to the purgatory if the two-step verification is enabled.
+            parameters = _purgatory.maybeAddToPurgatory(handler, requestParameter.parametersClass(), parameterConfigOverrides, _userTaskManager);
+        }
+
+        Request ccRequest = null;
+        if (parameters != null) {
+            requestConfigOverrides.put(requestParameter.parameterObject(), parameters);
+            ccRequest = _config.getConfiguredInstance(requestParameter.requestClass(), Request.class, requestConfigOverrides);
+        }
+
+        if (ccRequest != null) {
+            // ccRequest would be null if request is added to Purgatory.
+            ccRequest.handle(handler);
+        }
+    }
+
+    @Override
+    public AsyncKafkaCruiseControl asyncKafkaCruiseControl() {
+        return _asyncKafkaCruiseControl;
+    }
+
+    @Override
+    public Map<EndPoint, Timer> successfulRequestExecutionTimer() {
+        return Collections.unmodifiableMap(_successfulRequestExecutionTimer);
+    }
+
+    @Override
+    public ThreadLocal<Integer> asyncOperationStep() {
+        return _asyncOperationStep;
+    }
+
+    @Override
+    public UserTaskManager userTaskManager() {
+        return _userTaskManager;
+    }
+
+    @Override
+    public List<UserTaskManager.UserTaskInfo> getAllUserTasks() {
+        return _userTaskManager.getAllUserTasks();
+    }
+
+    @Override
+    public void kafkaClusterState(RoutingContext context) {
+        try {
+            doGetOrPost(context);
+        } catch (IOException e) {
+            e.printStackTrace();
```

Comment:
This isn't a nice way of handling the exceptions. Somehow we should tell vertx to return a HTTP 500 error code with the exception message. Also this logic could be extracted into a method instead of copy-pasting the same thing over all the api calls.